src/marshmallow/fields.py

```diff
@@ -487,29 +487,34 @@ def _nested_normalized_option(self, option_name):
             if field.startswith(nested_field)
         ]
 
-    def _serialize(self, nested_obj, attr, obj, **kwargs):
+    def _serialize(self, nested_obj, attr, obj, many=False, **kwargs):
         # Load up the schema first. This allows a RegistryError to be raised
         # if an invalid schema name was passed
         schema = self.schema
         if nested_obj is None:
             return None
         try:
-            return schema.dump(nested_obj, many=self.many)
+            many = self.many or many
+            return schema.dump(nested_obj, many=many)
         except ValidationError as exc:
             raise ValidationError(exc.messages, valid_data=exc.valid_data) from exc
 
-    def _test_collection(self, value):
-        if self.many and not utils.is_collection(value):
+    def _test_collection(self, value, many=False):
+        many = self.many or many
+        if many and not utils.is_collection(value):
             self.fail("type", input=value, type=value.__class__.__name__)
 
-    def _load(self, value, data, partial=None):
+    def _load(self, value, data, partial=None, many=False):
         try:
-            valid_data = self.schema.load(value, unknown=self.unknown, partial=partial)
+            many = self.many or many
+            valid_data = self.schema.load(
+                value, unknown=self.unknown, partial=partial, many=many
```

Comment:
I found it odd that `load()` wasn't actually passing `many` to `schema.load()` before, but `dump()` was. It turns out it will default to `schema.many`, which `Nested` sets when it builds its own schema instance. The `many` in `dump()` was redundant before, but it can be useful on both now.